edgelet/edgelet-docker/src/settings.rs

```diff
@@ -168,28 +164,33 @@ fn init_agent_spec(settings: &mut Settings) -> Result<(), LoadSettingsError> {
 
     // In nested scenario, Agent image can be pulled from its parent.
     // It is possible to specify the parent address using the keyword $upstream
-    agent_image_resolve(settings)?;
+    // agent_image_resolve(settings)?;
 
     Ok(())
 }
 
-fn agent_image_resolve(settings: &mut Settings) -> Result<(), LoadSettingsError> {
-    let image = settings.agent().config().image().to_string();
-
-    if let Some(parent_hostname) = settings.parent_hostname() {
-        if image.starts_with(UPSTREAM_PARENT_KEYWORD) {
-            let image_nested = format!(
-                "{}{}",
-                parent_hostname,
-                &image[UPSTREAM_PARENT_KEYWORD.len()..]
-            );
-            let config = settings.agent().config().clone().with_image(image_nested);
-            settings.agent_mut().set_config(config);
-        }
-    }
-
-    Ok(())
-}
+// TODO: wiring this up will be tricky, since Settings is loaded before the
+// parent_hostname is fetched from aziot...
+//
+// temporarily commenting this out to make forwards progress.
+
+// fn agent_image_resolve(settings: &mut Settings) -> Result<(), LoadSettingsError> {
```

Comment:
(I think you might be looking into this already - consider my suggestion if you haven't implemented)
The late binding for `$upstream` should be happening in docker config instead of edged - 
https://github.com/Azure/iotedge/blob/master/edgelet/edgelet-docker/src/config.rs#L12

This way, in the short to mid term future, when edge-agent logic makes its way down to edged, there won't be two different sources of truth to worry about. Current, there is coupling between the edged setting and the gatewayhostname environment variable - https://github.com/Azure/iotedge/blob/master/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/NestedEdgeParentUrlParser.cs#L8-L34